libraries/Update/src/HttpsOTAUpdate.cpp

```diff
@@ -0,0 +1,81 @@
+/* OTA task
+
+   This example code is in the Public Domain (or CC0 licensed, at your option.)
+
+   Unless required by applicable law or agreed to in writing, this
+   software is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+   CONDITIONS OF ANY KIND, either express or implied.
+*/
+#include <stdio.h>
+#include <string.h>
+
+#include <freertos/FreeRTOS.h>
+#include <freertos/task.h>
+
+#include "esp32-hal-log.h"
+#include "esp_http_client.h"
+#include "esp_https_ota.h"
+
+#include "HttpsOTAUpdate.h"
+#include "Esp.h"
+#define OTA_TASK_STACK_SIZE 9216
+
+typedef void (*HttpEventCb)(HttpEvent_t*);
+
+static esp_http_client_config_t config; 
+static HttpEventCb cb;
+static EventGroupHandle_t ota_status;//check for ota status
+static int check_status = -1;
+
+const int OTA_SUCCESSFUL_BIT = BIT0;
+const int OTA_FAILURE_BIT = BIT1;
+
+esp_err_t http_event_handler(esp_http_client_event_t *event)
+{
+    cb(event); 
+    return ESP_OK;
+}
+
+void https_ota_task(void *param)
+{
+    esp_err_t ret = esp_https_ota((const esp_http_client_config_t *)param);
+    if(ret == ESP_OK) {
+        xEventGroupSetBits(ota_status, OTA_SUCCESSFUL_BIT);
+    } else {
+        xEventGroupSetBits(ota_status, OTA_FAILURE_BIT);
+    }
+    vTaskDelete(NULL);
+}
+
+int HttpsOTAUpdateClass::status()
+{
+    EventBits_t set_bit =  xEventGroupWaitBits(ota_status, OTA_SUCCESSFUL_BIT | OTA_FAILURE_BIT, true, false, portMAX_DELAY);
+    if(set_bit & OTA_SUCCESSFUL_BIT) {
+        check_status = HTTPS_OTA_SUCCESS;
+    }
+    if(set_bit & OTA_FAILURE_BIT) {
+        check_status = HTTPS_OTA_FAIL;
+    }
+    xEventGroupClearBits(ota_status, OTA_SUCCESSFUL_BIT | OTA_FAILURE_BIT);    
```

Comment:
do you need to clear the bits here? maybe on `begin`? Also I see that you do not check if the event group is already created in begin. maybe you should check that and clear the bits if the group exists. Clearing here would mean that status can be asked only once, and that should not be a limitation IMO.

Reply:
Also I see that you are waiting for the bits. Is that necessary? Maybe status `UPDATING` makes sense?

Reply:
```cpp
typedef enum {
  HTTPS_OTA_IDLE,
  HTTPS_OTA_UPDATING,
  HTTPS_OTA_SUCCESS,
  HTTPS_OTA_FAIL
} HttpsOTAStatus_t;
```